src/gui/src/mainWindow.cpp

```diff
@@ -266,6 +266,49 @@ MainWindow::MainWindow(QWidget* parent)
   setWindowTitle("OpenROAD");
 }
 
+MainWindow::~MainWindow()
+{
+  // unregister descriptors
+  Gui::get()->unregisterDescriptor<Ruler*>();
+}
+
+void MainWindow::init(ord::OpenRoad* openroad, Tcl_Interp* interp)
```

Comment:
This looks to be the last instance of this problem.  Only db & sta are needed here.

Reply:
I could split this function into two parts, possibly, one that sets the DB and ensures openroad is initialized, and a second that actually uses the dbSta. The reason for passing in the openroad was to avoid this, since when `init` is called, openroad may or may not have been initialized and therefore I cannot use the dbSta pointer until afterwards.